diff --git a/.gitignore b/.gitignore
index ec376bb..c5461d6 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,26 @@
 .idea
-target
\ No newline at end of file
+target
+*.iml
+smilo.log
+.DS_Store
+
+# Compiled class file
+*.class
+
+# Log file
+*.log
+
+# BlueJ files
+*.ctxt
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
+
+# Package Files #
+*.jar
+*.war
+*.nar
+*.ear
+*.zip
+*.tar.gz
+*.rar
diff --git a/src/main/java/io/smilo/commons/peer/Peer.java b/src/main/java/io/smilo/commons/peer/Peer.java
index 8f214ed..c3ab63d 100644
--- a/src/main/java/io/smilo/commons/peer/Peer.java
+++ b/src/main/java/io/smilo/commons/peer/Peer.java
@@ -53,13 +53,13 @@ public class Peer implements Runnable, IPeer {
         this.initialized = false;
         this.address = socket.getInetAddress();
         this.remotePort = socket.getPort();
-        this.identifier = identifier;
+        this.identifier = (identifier != "") ? identifier : socket.getInetAddress().toString();
         setKeepAlive();
     }
 
     public Peer(String identifier, InetAddress address, int port) throws IOException {
         this(identifier, new Socket(address, port));
-        this.identifier = identifier;
+        this.identifier = (identifier != "") ? identifier : address.toString()+ ":" +port;
     }
 
     public Peer() {
diff --git a/src/main/java/io/smilo/commons/peer/PeerReceiver.java b/src/main/java/io/smilo/commons/peer/PeerReceiver.java
index b8c029f..bb3700a 100644
--- a/src/main/java/io/smilo/commons/peer/PeerReceiver.java
+++ b/src/main/java/io/smilo/commons/peer/PeerReceiver.java
@@ -93,7 +93,7 @@ public class PeerReceiver {
                         LOGGER.error("Incomplete message... " + data);
                         peer.write("ERROR the message was not complete!");
                     } catch (Exception e) {
-                        LOGGER.error("No idea what happened here...", e);
+                        LOGGER.error("No idea what happened here... MSG: " + data, e);
                     }
                 }
             });
